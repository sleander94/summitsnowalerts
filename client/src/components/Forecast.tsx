import { useState, useEffect } from 'react';
import { AuthProps, mountainsObj } from '../types.d';
import MountainWeather from './MountainWeather';

const Forecast = ({ user }: AuthProps) => {
  const [mountains, setMountains] = useState<mountainsObj>();

  const getMountains = async () => {
    try {
      const response = await fetch('/users', {
        method: 'GET',
        headers: {
          'Content-Type': 'application/json',
        },
      });
      const data = await response.json();
      setMountains(data.mountains);
      setContent(true);
    } catch (err) {
      console.error(err);
    }
  };

  useEffect(() => {
    getMountains();
  }, [user]);

  const [content, setContent] = useState<boolean>(false);

  return (
    <section id="conditions">
      <div className="background-image"></div>
      <div className="content">
        <h1>My Mountain Forecasts</h1>
        {!user && (
          <h2 className="no-user">
            Log in to view the weather at your mountains.
          </h2>
        )}
        <div className="weather">
          {mountains &&
            Object.keys(mountains).map((key) => {
              const mountain = key as keyof mountainsObj;
              return (
                <MountainWeather
                  key={mountain}
                  name={mountain}
                  location={mountains[mountain] as number}
                />
              );
            })}
        </div>
      </div>
    </section>
  );
};

export default Forecast;
