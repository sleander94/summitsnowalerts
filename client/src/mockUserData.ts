export const mockUserData = {
  _id: 'aoifh73$%D3g3',
  name: 'Test User',
  email: 'test@email.com',
  phone: '(555)555-5555',
  password: 'aorinferu142431e5e23e',
  notifications: {
    text: true,
    email: false,
    times: {
      0: true,
      1: false,
      2: false,
      3: false,
      4: false,
      5: false,
      6: false,
      7: false,
      8: true,
      9: false,
      10: false,
      11: false,
      12: false,
      13: false,
      14: false,
      15: false,
      16: false,
      17: false,
      18: true,
      19: false,
      20: false,
      21: false,
      22: false,
      23: false,
    },
    days: {
      sunday: true,
      monday: false,
      tuesday: false,
      wednesday: false,
      thursday: true,
      friday: true,
      saturday: true,
    },
  },
  mountains: { Breckenridge: 80424, Keystone: 80435 },
};
