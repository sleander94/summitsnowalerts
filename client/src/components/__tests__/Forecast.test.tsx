import Forecast from '../Forecast';
import { render, screen, waitFor } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import { mockUserData } from '../../mockUserData';

const server = setupServer(
  rest.get('/users', (req, res, ctx) => {
    return res(ctx.json(mockUserData));
  }),
  rest.get('https://api.weatherapi.com/v1/forecast.json', (req, res, ctx) => {
    return res(
      ctx.json({
        today: {
          condition: 'Sunny',
          icon: 'fakeurl',
          avgtemp_f: '72.3',
          daily_chance_of_snow: 0,
          totalprecip_in: 0,
        },
        tomorrow: {
          condition: 'Snowy',
          icon: 'fakeurl',
          avgtemp_f: '28',
          daily_chance_of_snow: 0.95,
          totalprecip_in: 5.5,
        },
      })
    );
  })
);

beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());

it('Displays login message when logged out', () => {
  render(
    <Router>
      <Forecast user={undefined} />
    </Router>
  );

  expect(
    screen.getByText('Log in to view the weather at your mountains.')
  ).toBeInTheDocument();
  expect(screen.queryByText('Breckenridge')).not.toBeInTheDocument();
});

it("Displays MountainWeather components for user's mountains when logged in", async () => {
  render(
    <Router>
      <Forecast user={mockUserData} />
    </Router>
  );

  await waitFor(() =>
    expect(document.getElementById('Breckenridge')).toBeInTheDocument()
  );
  await waitFor(() =>
    expect(document.getElementById('Keystone')).toBeInTheDocument()
  );
});
