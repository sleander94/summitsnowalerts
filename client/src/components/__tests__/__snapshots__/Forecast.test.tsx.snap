// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Displays login prompt message when logged out 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        id="forecast"
      >
        <div
          class="background-image"
        />
        <div
          class="content"
        >
          <h1>
            My Mountain Forecasts
          </h1>
          <h2
            class="no-user"
          >
            Log in to view the weather at your mountains.
          </h2>
          <div
            class="weather"
          />
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      id="forecast"
    >
      <div
        class="background-image"
      />
      <div
        class="content"
      >
        <h1>
          My Mountain Forecasts
        </h1>
        <h2
          class="no-user"
        >
          Log in to view the weather at your mountains.
        </h2>
        <div
          class="weather"
        />
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Renders correctly after loading weather data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        id="forecast"
      >
        <div
          class="background-image"
        />
        <div
          class="content"
        >
          <h1>
            My Mountain Forecasts
          </h1>
          <div
            class="weather"
          >
            <div
              class="mountain-weather"
              id="Breckenridge"
            >
              <div
                class="weather-content"
              >
                <h2>
                  Breckenridge
                </h2>
                <div
                  class="current-conditions"
                >
                  <h3>
                    Today
                  </h3>
                  <p>
                    Sunny
                  </p>
                  <img
                    alt="Sunny icon"
                    src="fakeurl"
                  />
                  <p>
                    72.3
                     °F
                  </p>
                  <p>
                    Snow: 
                    0
                    %
                  </p>
                  <p>
                    Precip: 
                    0
                     in.
                  </p>
                </div>
                <div
                  class="forecast"
                >
                  <h3>
                    Tomorrow
                  </h3>
                  <p>
                    Snowy
                  </p>
                  <img
                    alt="Snowy icon"
                    src="fakeurl"
                  />
                  <p>
                    28
                     °F
                  </p>
                  <p>
                    Snow: 
                    95
                    %
                  </p>
                  <p>
                    Precip: 
                    5.5
                     in.
                  </p>
                </div>
              </div>
            </div>
            <div
              class="mountain-weather"
              id="Keystone"
            >
              <div
                class="weather-content"
              >
                <h2>
                  Keystone
                </h2>
                <div
                  class="current-conditions"
                >
                  <h3>
                    Today
                  </h3>
                  <p>
                    Sunny
                  </p>
                  <img
                    alt="Sunny icon"
                    src="fakeurl"
                  />
                  <p>
                    72.3
                     °F
                  </p>
                  <p>
                    Snow: 
                    0
                    %
                  </p>
                  <p>
                    Precip: 
                    0
                     in.
                  </p>
                </div>
                <div
                  class="forecast"
                >
                  <h3>
                    Tomorrow
                  </h3>
                  <p>
                    Snowy
                  </p>
                  <img
                    alt="Snowy icon"
                    src="fakeurl"
                  />
                  <p>
                    28
                     °F
                  </p>
                  <p>
                    Snow: 
                    95
                    %
                  </p>
                  <p>
                    Precip: 
                    5.5
                     in.
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      id="forecast"
    >
      <div
        class="background-image"
      />
      <div
        class="content"
      >
        <h1>
          My Mountain Forecasts
        </h1>
        <div
          class="weather"
        >
          <div
            class="mountain-weather"
            id="Breckenridge"
          >
            <div
              class="weather-content"
            >
              <h2>
                Breckenridge
              </h2>
              <div
                class="current-conditions"
              >
                <h3>
                  Today
                </h3>
                <p>
                  Sunny
                </p>
                <img
                  alt="Sunny icon"
                  src="fakeurl"
                />
                <p>
                  72.3
                   °F
                </p>
                <p>
                  Snow: 
                  0
                  %
                </p>
                <p>
                  Precip: 
                  0
                   in.
                </p>
              </div>
              <div
                class="forecast"
              >
                <h3>
                  Tomorrow
                </h3>
                <p>
                  Snowy
                </p>
                <img
                  alt="Snowy icon"
                  src="fakeurl"
                />
                <p>
                  28
                   °F
                </p>
                <p>
                  Snow: 
                  95
                  %
                </p>
                <p>
                  Precip: 
                  5.5
                   in.
                </p>
              </div>
            </div>
          </div>
          <div
            class="mountain-weather"
            id="Keystone"
          >
            <div
              class="weather-content"
            >
              <h2>
                Keystone
              </h2>
              <div
                class="current-conditions"
              >
                <h3>
                  Today
                </h3>
                <p>
                  Sunny
                </p>
                <img
                  alt="Sunny icon"
                  src="fakeurl"
                />
                <p>
                  72.3
                   °F
                </p>
                <p>
                  Snow: 
                  0
                  %
                </p>
                <p>
                  Precip: 
                  0
                   in.
                </p>
              </div>
              <div
                class="forecast"
              >
                <h3>
                  Tomorrow
                </h3>
                <p>
                  Snowy
                </p>
                <img
                  alt="Snowy icon"
                  src="fakeurl"
                />
                <p>
                  28
                   °F
                </p>
                <p>
                  Snow: 
                  95
                  %
                </p>
                <p>
                  Precip: 
                  5.5
                   in.
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
